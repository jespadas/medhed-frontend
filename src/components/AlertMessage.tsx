import * as React from 'react';
import Alert from '@mui/material/Alert';
import Stack from '@mui/material/Stack';

export const AlertMessage = () => {
  return (
    <Stack sx={{ width: '100%', display: 'flex', justifyContent: 'center', height:'100vh', alignItems: 'center'}}>
      <Alert severity="error"><strong>Error Loading </strong> - There was an error loading your data. Please try again later or contact the network administrator.</Alert>
    </Stack>
  );
}