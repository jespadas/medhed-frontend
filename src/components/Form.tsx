import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import Paper from '@mui/material/Paper';
import Box from '@mui/material/Box';
import AdapterDateFns from '@mui/lab/AdapterDateFns';
import DesktopDatePicker from '@mui/lab/DesktopDatePicker';
import LocalizationProvider from '@mui/lab/LocalizationProvider';
import TextField from '@mui/material/TextField';
import Typography from '@mui/material/Typography';
import Select from '@mui/material/Select';
import InputLabel from '@mui/material/InputLabel';
import Button from '@mui/material/Button';
import FormControl from '@mui/material/FormControl';
import MenuItem from '@mui/material/MenuItem';
import { CreateReservationRest } from '../models/reservationModel';
import { ShiftModel } from '../models/shiftModel';
import { createReservation } from '../api/hospital';

const currentDate = new Date();
const initReservation: CreateReservationRest = {
	date: currentDate.toISOString(),
	hospitalId: 0,
	patient: 0,
	shiftId: 0,
};

interface Props {
	shifts: ShiftModel[];
}

export const Form = ({ shifts }: Props) => {
	const { idHospital } = useParams<any>();
	const [editedReservation, setEditedReservation] =
		useState<CreateReservationRest>(initReservation);

	const [isAccepted, setIsAccepted] = useState(false);

	const handleChangeShift = (e: any) => {
		console.log(e);
		
		setEditedReservation({
			...editedReservation,
			shiftId: e.target.value,
		});
	};

	useEffect(() => {
		const setIdHospital = () => {
			setEditedReservation({
				...editedReservation,
				hospitalId: Number(idHospital),
			});
		};

		setIdHospital();
		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, []);

	const isInvalid = {
		date: (isAccepted && (editedReservation.date === null)) || editedReservation.date?.toString() === 'Invalid Date',
		hospitalId: isAccepted && editedReservation.hospitalId === 0,
		patient: isAccepted && editedReservation.patient === 0,
		shiftId: isAccepted && editedReservation.shiftId === 0,
	}

	const isFormValid = () => {
		const isValid = Object.values(isInvalid).findIndex(item => item) === -1;
		return isValid;
	}
	
	const saveReservation = async () => {
		try {
			const response = await createReservation(editedReservation);
			console.log(response);
			
		} catch (error) {
			console.log(error);
			
		}
	};

	const handleSave = () => {
		setIsAccepted(true);
		if (isFormValid()) {
			saveReservation();
		}
	}

	return (
		<Paper sx={{ padding: '2em' }}>
			<Box
				component='form'
				sx={{
					'& > :not(style)': { m: 1, width: '25ch' },
				}}
				noValidate
				autoComplete='off'
			>
				<Typography gutterBottom variant='h5' component='h2'>
					Make an ER reservation
				</Typography>
				<TextField
					id='standard-basic'
					label='Hospital'
					variant='outlined'
					value={idHospital ? idHospital : editedReservation.hospitalId}
					disabled
					onChange={(e) =>
						setEditedReservation({
							...editedReservation,
							hospitalId: Number(e.target.value),
						})
					}
					error={isInvalid.hospitalId}
				/>
				<TextField
					type='number'
					InputProps={{
						inputProps: {
							max: 3,
							min: 0,
						},
					}}
					id='standard-basic'
					label='Patient'
					variant='outlined'
					value={editedReservation.patient}
					onChange={(e) => {
						setEditedReservation({
							...editedReservation,
							patient: e.target.value !== '' ? Number(e.target.value) : '',
						});
					}}
					error={isInvalid.patient}
				/>
				<LocalizationProvider dateAdapter={AdapterDateFns}>
					<DesktopDatePicker
						disablePast
						label='Date'
						inputFormat='MM/dd/yyyy'
						value={editedReservation.date}
						onChange={(date: any) => {
							setEditedReservation({
								...editedReservation,
								date,
							});
						}}
						renderInput={(params) => <TextField {...params} error={isInvalid.date}/>}
					/>
				</LocalizationProvider>
				<FormControl sx={{ m: 1, minWidth: 120 }}>
					<InputLabel id='demo-simple-select-helper-label'>Shift</InputLabel>
					<Select
						labelId='demo-simple-select-helper-label'
						id='demo-simple-select-helper'
						value={editedReservation.shiftId}
						label='Shift'
						onChange={handleChangeShift}
						error={isInvalid.shiftId}
					>
						<MenuItem value={0}>Select a shift</MenuItem>
						{shifts.map((shift) => (
							<MenuItem key={shift.id} value={shift.id}>
								{shift.name}
							</MenuItem>
						))}
					</Select>
				</FormControl>
			</Box>
			<div
				style={{
					display: 'flex',
					justifyContent: 'flex-end',
					paddingTop: '1em',
				}}
			>
				<Button variant='contained' onClick={handleSave}>Make the reservaton</Button>
			</div>
		</Paper>
	);
};
