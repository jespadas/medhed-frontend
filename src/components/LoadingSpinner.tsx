import * as React from 'react';
import CircularProgress from '@mui/material/CircularProgress';
import Box from '@mui/material/Box';

export const LoadingSpinner = () => {
  return (
    <Box sx={{ display: 'flex', justifyContent: 'center', height:'100vh', alignItems: 'center'}}>
      <CircularProgress sx={{ size: 300 }}/>
    </Box>
  );
}