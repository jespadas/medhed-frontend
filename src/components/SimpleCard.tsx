import * as React from 'react';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import { Link } from 'react-router-dom';

interface Props {
	id: number
    title: string;
    description: string;
    image: string;
    handleButton: () => void;
	to: any
}

export const SimpleCard = ({title, description, image, handleButton,to} : Props) => {
	return (
		<Card sx={{ minWidth: 200 }}>
			<CardMedia
				component='img'
				height='140'
				image={image}
				alt='hospital'
                title={title}
			/>
			<CardContent>
				<Typography gutterBottom variant='h5' component='div' sx={{color: "white"}}>
					{title}
				</Typography>
				<Typography variant='body2' color='text.secondary'>
					{description}
				</Typography>
			</CardContent>
			<CardActions>
				<Link to={to} >
				<Button size='small' onClick={handleButton}>Login</Button>
				</Link>
			</CardActions>
		</Card>
	);
}
