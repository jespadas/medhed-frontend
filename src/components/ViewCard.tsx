import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import CardActionArea from '@mui/material/CardActionArea';
import * as React from 'react';

interface Props {
	id: number;
	title: string;
	description: string;
	image: string;
	subDescription: any;
}

const styles = {
	media: {
		height: '100%',
	},
};

export const ViewCard = ({
	title,
	description,
	image,
	subDescription,
}: Props) => {
	return (
		<Card sx={{ minWidth: 250, padding: '1em' }}>
			<CardActionArea>
				<CardMedia
					//sx={{ height: '70vh' }}
					classes={styles.media}
					component='img'
					image={image}
					alt='hospital'
					title={title}
				/>
				<CardContent>
					<Typography gutterBottom variant='h5' component='h2'>
						{title}
					</Typography>
					<Typography variant='body1' color='text.secondary' component='p'>
						{description}
					</Typography>
					<Typography variant='body1' color='text.secondary' component='p'>
						{subDescription}
					</Typography>
				</CardContent>
			</CardActionArea>
		</Card>
	);
};
