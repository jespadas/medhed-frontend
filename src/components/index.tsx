export * from './SimpleCard';
export * from './ManageStatus';
export * from './ViewCard';
export * from './Form';