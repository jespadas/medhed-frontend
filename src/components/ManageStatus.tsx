import { ReactElement } from 'react';
import Box from '@mui/material/Box';
import { LoadingSpinner } from '../components/LoadingSpinner';
import { AlertMessage } from '../components/AlertMessage';

interface Props {
	status: 'loading' | 'error' | 'success';
	children: ReactElement;
}

export const ManageStatus = ({ status, children }: Props) => {
	if (status === 'loading') {
		return <LoadingSpinner />;
	}

	if (status === 'error') {
		return <AlertMessage />;
	}

	if (status === 'success') {
		return <Box>{children}</Box>;
	}

    return null;
};
