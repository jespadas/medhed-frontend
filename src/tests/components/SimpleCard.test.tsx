import React from 'react';
import { shallow } from 'enzyme';
import { SimpleCard } from '../../components/SimpleCard';

describe('Test in <SimpleCard />', () => {

    const title = "the hospital title";

    const url = "https://localhost:3000/img.png";

    const handleButton = ()=>{};

    const description = 'this is the hospital description';

    const id = 1;

    const to = 'https://localhost:3000/img.png'

    const wrapper = shallow(<SimpleCard title={title} image={url} handleButton={handleButton} description={description} id={id} to={to} />);

    test('Should display the component', () => {

        expect(wrapper).toMatchSnapshot();

    })

})