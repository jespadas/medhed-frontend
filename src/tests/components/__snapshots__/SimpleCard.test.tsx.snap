// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SimpleCard /> Should display the component 1`] = `
<ForwardRef(Card)
  sx={
    Object {
      "minWidth": 200,
    }
  }
>
  <ForwardRef(CardMedia)
    alt="hospital"
    component="img"
    height="140"
    image="https://localhost:3000/img.png"
    title="the hospital title"
  />
  <ForwardRef(CardContent)>
    <ForwardRef(Typography)
      component="div"
      gutterBottom={true}
      sx={
        Object {
          "color": "white",
        }
      }
      variant="h5"
    >
      the hospital title
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      color="text.secondary"
      variant="body2"
    >
      this is the hospital description
    </ForwardRef(Typography)>
  </ForwardRef(CardContent)>
  <ForwardRef(CardActions)>
    <Link
      to="https://localhost:3000/img.png"
    >
      <ForwardRef(Button)
        onClick={[Function]}
        size="small"
      >
        Login
      </ForwardRef(Button)>
    </Link>
  </ForwardRef(CardActions)>
</ForwardRef(Card)>
`;
